### PR TITLE
[AMORO-1959] fix orphan files clean mistakenly delete the statistics files

### LINKED ISSUE #1959: [Bug]: Orphan files clean mistakenly deleted the Iceberg statistic file
### What happened?

When orphan-files-clean service clean the metadata files, it mistakenly deleted the Iceberg statistic file.

### Affects Versions

0.5.0

### What engines are you seeing the problem on?

AMS

### How to reproduce

1.create an iceberg format table
2.create a snapshot and write a statistic file to the table
3.open orphan files clean service of the table
4.after clean orphan files, the statistic file will be deleted from the file system

### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

Close #1959 .

## Brief change log

- orphan files clean should not clean statistics files
- add unit test case

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution.

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/OrphanFilesCleaningExecutor.java [hunk 1]
```diff
@@ -325,9 +325,12 @@ private static Set<String> getValidMetadataFiles(UnkeyedTable internalTable) {
           cnt,
           size);
     }
-    Stream.concat(
+    Stream.of(
             ReachableFileUtil.metadataFileLocations(internalTable, false).stream(),
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/table/executor/OrphanFilesCleaningExecutor.java:329]: Amoro project uses 2 spaces as indent, can you fix the format here?

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/OrphanFilesCleaningExecutor.java:329]: OK, I will check it.
